File: C++/0167_twosumiiinputarrayissorted.cc
// 2020-08-04 01:30:00

#include <vector>

using std::vector;

class Solution
{
public:
    vector<int> twoSum(vector<int> &numbers, int target)
    {
        if (numbers.size() <= 1)
            return {};
        int i = 0, j = numbers.size() - 1;
        while (i < j)
        {
            if (i < j && target > numbers[i] + numbers[j])
                ++i;
            else if (i < j && target < numbers[i] + numbers[j])
                --j;
            else
                return {i + 1, j + 1};
        }
        return {};
    }
};

File: Python/0167_two_sum_ii_input_array_is_sorted.py
# 2020-08-04 00:03:10

from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if numbers is None or len(numbers) <= 1:
            return [0, 0]
        i = 0
        j = len(numbers) - 1
        while i < j:
            while i < j and target > numbers[j] + numbers[i]:
                i += 1
            while i < j and target < numbers[j] + numbers[i]:
                j -= 1
            if target == numbers[j] + numbers[i]:
                return [i + 1, j + 1]
            else:
                continue
        return [0, 0]
